freqtrade/persistence/models.py

```diff
@@ -504,6 +514,64 @@ def update_trade(self, order: Order) -> None:
             raise ValueError(f'Unknown order type: {order.order_type}')
         Trade.commit()
 
+    def process_sell_sub_trade(self, order: Order, is_closed: bool = True,
+                               is_non_bt: bool = True) -> None:
+        orders = (self.select_filled_orders('buy'))
+        sell_amount = order.safe_filled
+        sell_rate = order.safe_price
+        sell_stake_amount = sell_rate * sell_amount * (1 - self.fee_close)
+        if is_closed:
+            if sell_amount == self.amount:
+                self.close(sell_rate)
+                if is_non_bt:
+                    Trade.commit()
+                return
+        realized_profit = 0.0
+        profit = 0.0
+        idx = -1
+        while sell_amount:
+            b_order = orders[idx]
+            buy_amount = b_order.safe_amount_after_fee
+            avg_rate = b_order.safe_price
+            buy_rate = b_order.initial_average or b_order.price
+            if sell_amount < buy_amount:
+                amount = sell_amount
+            else:
+                idx -= 1
+                amount = buy_amount
+            if is_closed:
+                b_order.filled -= amount
+                b_order.order_update_date = datetime.now(timezone.utc)
+            sell_amount -= amount
+            profit += self.calc_profit2(avg_rate, sell_rate, amount)
+            realized_profit += self.calc_profit2(buy_rate, sell_rate, amount)
+        if is_closed:
+            b_order2 = orders[idx]
+            amount2 = b_order2.safe_amount_after_fee
+            b_order2.average = (b_order2.average * amount2 - profit / (1 + self.fee_open)) / amount2
```

Comment:
the average corresponds to the **order's average** - how it filled on the exchange - and the same is true for the whole order object.
Therefore this is simply wrong - the average fill price of the order will **NEVER** change based on other orders.

Reply:
This statements wil only run once when the sell order is closed.

Reply:
Let me repeat what i said above (and in the change statement).

The order object is not to be updated - it's there to reflect the state of the order on the exchange.

It does therefore not matter when or how often this code is running - it's modifying fields in the database it should not.

the **order's average price did not change** - and it never will - as it's an Order on the exchange.

This is not something to argue about - it's a simple fact and i will not accept anything that will change this.